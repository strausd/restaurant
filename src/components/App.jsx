import React from 'react';
import { connect } from 'react-redux';

import AppRouter from '../routers/AppRouter';
import { fetchRestaurants } from '../actions/baseActions'


class App extends React.Component {
    constructor(props) {
        super(props);
        props.fetchRestaurants();
    }

    render() {
        return (
            <div>
                <AppRouter />
            </div>
        );
    }
}

const mapDispatchToProps = dispatch => {
    return {
        fetchRestaurants: () => dispatch(fetchRestaurants())
    };
};

export default connect(undefined, mapDispatchToProps)(App);