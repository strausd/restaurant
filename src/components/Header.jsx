import React from 'react';
import { Link } from 'react-router-dom';


const Header = props => {
    const back_button = window.location.pathname !== '/';
    return (
        <div className="header">
            {back_button ? (
                <div className="header__icon header__back">
                    <Link to="/">
                        <img src="/dist/icons/ic_webBack@2x.png" alt="Map" />
                    </Link>
                </div>
            ) : null}
            <div className="header__link">
                <Link to="/">
                    <h1>Lunch Tyme</h1>
                </Link>
            </div>
            <div className="header__icon header__map">
                <Link to="/map">
                    <img src="/dist/icons/icon_map@2x.png" alt="Map" />
                </Link>
            </div>
        </div>
    );
};

export default Header;