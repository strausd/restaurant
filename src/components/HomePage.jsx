import React from 'react';
import { connect } from 'react-redux'

import RestaurantsList from './RestaurantsList';


export const HomePage = () => {
    return (
        <div className="page-content">
            <RestaurantsList />
        </div>
    );
};

export default HomePage;