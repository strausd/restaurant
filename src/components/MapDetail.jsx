import React from 'react';
import { withScriptjs, withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps"

const MapDetail = withScriptjs(withGoogleMap((props) =>
    <GoogleMap
        defaultZoom={props.zoom}
        defaultCenter={{ lat: props.lat, lng: props.lng }}
    >
        <Marker position={{ lat: props.lat, lng: props.lng }}>
            <InfoWindow>
                <div>Testing</div>
            </InfoWindow>
        </Marker>
    </GoogleMap>
));

export default MapDetail;