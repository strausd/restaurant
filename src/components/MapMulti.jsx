import React from 'react';
import { withScriptjs, withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps";

export const MapDetail = withScriptjs(withGoogleMap((props) =>
    <GoogleMap
        defaultZoom={props.zoom}
        defaultCenter={{ lat: props.lat, lng: props.lng }}
    >
        {props.restaurants.map((r, index) => {
            return (
                <Marker key={index} position={{ lat: r.lat, lng: r.lng }} onClick={() => props.onMarkerToggle(index)}>
                    {r.isOpenMarkerInfo && <InfoWindow onCloseClick={props.onMarkerToggle}>
                        <div className="marker-info">
                            <div className="marker-info__name">{r.name}</div>
                            <div className="marker-info__directions">
                                <a target="_blank" href={'https://www.google.com/maps/dir/?api=1&destination=' + r.gMapsQuery}>
                                    Get Directions
                                </a>
                            </div>
                        </div>
                    </InfoWindow>}
                </Marker>
            );
        })}
    </GoogleMap>
));

export default MapDetail;