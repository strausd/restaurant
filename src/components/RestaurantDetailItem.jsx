import React from 'react';

// AIzaSyCRGU4zvepa0suuU1BZu_tWiEn7ho5MSEc
const RestaurantDetailItem = ({ restaurant }) => {
    const destination_str = encodeURIComponent(
        restaurant.name + ' ' +
        restaurant.location.address + ' ' +
        restaurant.location.city + ' ' +
        restaurant.location.state + ' ' +
        restaurant.location.postalCode
    );
    return (
        <div className="restaurant-detail">
            <div className="restaurant-detail__map">

            </div>
            <div className="restaurant-detail__header">
                <h2>{restaurant.name}</h2>
                <div className="restaurant-detail__category">{restaurant.category}</div>
            </div>
            <div className="restaurant-detail__info">
                <div className="restaurant-detail__address">
                    <a target="_blank" href={'https://www.google.com/maps/dir/?api=1&destination=' + destination_str}>
                        {restaurant.location.address}
                        <br/>
                        {restaurant.location.city}, {restaurant.location.state} {restaurant.location.postalCode}
                    </a>
                </div>
                {
                    !restaurant.contact ? null : (
                        <div>
                            <div className="restaurant-detail__phone">
                                <a href={'tel:' + restaurant.contact.phone} >{restaurant.contact.formattedPhone}</a>
                            </div>
                            <div className="restaurant-detail__twitter">
                                <a href={'https://twitter.com/' + restaurant.contact.twitter} target="_blank" >@{restaurant.contact.twitter}</a>
                            </div>
                        </div>
                    )
                }
            </div>
        </div>
    );
};

export default RestaurantDetailItem;