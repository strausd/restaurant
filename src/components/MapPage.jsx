import React from 'react';
import { connect } from 'react-redux';

import MapMulti from './MapMulti';
import { getMapInfoArray } from '../misc/utils';


export class MapPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            restaurant_marker_info: []
        };
    }

    componentWillMount() {
        if (!this.props.restaurants || this.props.restaurants.length < 1) {
            this.setState({ restaurant_marker_info: [] });
        } else {
            const restaurants = getMapInfoArray(this.props.restaurants);
            this.setState({ restaurant_marker_info: restaurants });
        }
    }
    

    componentWillReceiveProps(nextProps) {
        if (!nextProps.restaurants || nextProps.restaurants.length < 1) {
            this.setState({ restaurant_marker_info: [] });
        } else {
            const restaurants = getMapInfoArray(nextProps.restaurants);
            this.setState({ restaurant_marker_info: restaurants });
        }
    }

    onMarkerToggle = (index) => {
        this.setState(prevState => {
            return {
                restaurant_marker_info: prevState.restaurant_marker_info.map((r, r_index) => {
                    if (index === r_index) {
                        return {
                            ...r,
                            isOpenMarkerInfo: !r.isOpenMarkerInfo
                        };
                    } else {
                        return {
                            ...r,
                            isOpenMarkerInfo: false
                        };
                    }
                })
            };
        });
    }

    render() {
        if (!this.props.restaurants || this.props.restaurants.length < 1) {
            return (
                <div className="page-content">
                    Loading...
                </div>
            );
        } else {
            return (
                <div className="page-content">
                    <div className="map-page">
                        <MapMulti 
                            isMarkerShow={true}
                            googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyBGyVswgcCkctFs8x4X-14BS16VXWINHPA&v=3.exp&libraries=geometry,drawing,places"
                            loadingElement={<div style={{ height: `100%` }} />}
                            containerElement={<div style={{ height: `100%` }} />}
                            mapElement={<div style={{ height: `100%` }} />}
                            lat={32.98}
                            lng={-96.835}
                            zoom={13}
                            onMarkerToggle={this.onMarkerToggle}
                            restaurants={this.state.restaurant_marker_info}
                        />
                    </div>
                </div>
            );
        }
        
    }
}

const mapStateToProps = ({ restaurants }) => {
    return {
        restaurants
    };
};

export default connect(mapStateToProps)(MapPage);