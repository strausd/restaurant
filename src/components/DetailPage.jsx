import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import RestaurantDetailItem from './RestaurantDetailItem';


export class DetailPage extends React.Component {
    render() {
        const id = this.props.match.params.id;
        if (isNaN(id)) {
            return <Redirect to="/" />
        } else if (!this.props.restaurants || this.props.restaurants.length < 1) {
            return (
                <div className="page-content">
                    Loading...
                </div>
            );
        } else {
            const restaurant = this.props.restaurants[id];
            return (
                <div className="page-content">
                    <RestaurantDetailItem restaurant={restaurant} />
                </div>
            );
        }
    }
}


const mapStateToProps = ({ restaurants, restaurant }) => {
    return {
        restaurants,
        restaurant
    };
};

export default connect(mapStateToProps)(DetailPage);