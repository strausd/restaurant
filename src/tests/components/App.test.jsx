import React from 'react';
import { shallow } from 'enzyme';

import { App } from '../../components/App';

let wrapper, fetchRestaurants;

beforeAll(() => {
    fetchRestaurants = jest.fn();
    wrapper = shallow(<App fetchRestaurants={fetchRestaurants} />);
});

test('should render App component properly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(fetchRestaurants).toHaveBeenCalledTimes(1);
});