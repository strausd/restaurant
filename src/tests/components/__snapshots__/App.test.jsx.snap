// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchRestaurants={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <AppRouter />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <AppRouter />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
